File: cash.py
from cs50 import get_float


def main():
    # get change from user and ensure it is a positive number
    while True:
        change = get_float("how much change is owed: ")
        if change > 0:
            break

    # remove the decimal and any imprecision
    change = round(change * 100)
    count = 0

    # check for each coin
    while change > 24:
        # quarters
        change -= 25
        count += 1
    while change > 9:
        # dimes
        change -= 10
        count += 1
    while change > 4:
        # nickels
        change -= 5
        count += 1
    while change > 0:
        # pennies
        change -= 1
        count += 1

    print(count)


if __name__ == "__main__":
    main()

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    //prompt user for amount of change owed in format xx.xx
    float c = 0;
    do
    {
        c = get_float("How much change is owed? Input: ");        
    }
    while (c < 0);
    
    //convert float to int and total number of cents
    //declare coin variables
    int total = round(c * 100), q = 0, d = 0, n = 0, p = 0;
    
    //if there are greater than 25 cents, add one to q and subtract 25 from total
    while (total > 24)
    {
        total = total - 25;
        q++;
    }
    
    //run the same for each coin size
    while (total > 9)
    {
        total = total - 10;
        d++;
    }


    while (total > 4)
    {
        total = total - 5;
        d++;
    }

     while (total > 0)
    {
        total = total - 1;
        p++;
    }
   
    
    int coins = q + d + n + p;
    printf("%i\n", coins);
}
